qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -232,6 +218,12 @@ def _solve_internal(
             eigen_result = self._min_eigen_solver.compute_minimum_eigenvalue(operator)
             # analyze results
             raw_samples = None
+            if not hasattr(eigen_result, "eigenstate"):
+                raise QiskitOptimizationError(
+                    "MinimumEigenOptimizer does not support this VQE "
+                    f"{type(self._min_eigen_solver)}. "
+                    "You can use qiskit_algorithms.SamplingVQE instead."
+                )
```

Comment:
Instead of early type check of `SamplingVQE` or `NumPyMinimumEigensolver`, I added a result check when `solve` is invoked.